### PR TITLE
#1 Added functionality for arbitrary database columns

### PR DESCRIPTION
**Pull request by Trevortds**
Email is still required. Also added option to name the table at the maintainer's request. If the table name is used twice, the user is asked if they want to append or overwrite.

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -33,17 +33,25 @@ def stop(delete):
...
     try:
         file = read_file(file_name)
-    except:
-        click.echo('Abort. Incorrect file format.')
+    except Exception as e:
+        click.echo(f'Abort. Incorrect file format. {str(e)}')
         return None
     engine = connect_db()
-    not_loaded = load_file_to_db(file, engine)
+    if 'users' in engine.table_names():
```
Comment (Cheukting) [inline app/app.py]: I think it should be `name` instead of `'users'` ?

File: app/util.py [hunk 2]
```diff
@@ -56,31 +65,36 @@ def delete_db():
 def read_file(file_name):
     """Given the path of a file, it will be loaded and returned as DataFrame
     Duplicates of emails will be altomatically removed"""
-
-    header = ['firstname','lastname','email']
+    with open(file_name, "r") as f:
```
Comment (Cheukting) [inline app/util.py]: I think we can use pandas to read the csv (pd.read_csv) and get the column names from there, if no key column(s) is specified by the user, we will use the index (which will be just 0,1,2,3,4... by default) of the pandas DataFrame as a key. We can name it `default_id`.

**Review by Cheukting — COMMENTED** [review]
Very nice work! Thank you. Beside one spot that I think `'users'` should be `name`, I have more though regarding the key of the table. Now the key is set to `email` no matter what,  I think this contrain should also be remove. Maybe we can provide aother option `--keys` to specify key column(s) (just like `--name`), if nothing is specified, we will just use the default index of the pandas DataFrame as a key (named `default_id`)

Sorry for requesting more work about the key. If you don't feel like implement it in this PR, let me know so I will merge your great work so far and we will improve it in another issue.

File: app/app.py (hunk 1 above)
Comment (Trevortds) [reply app/app.py]: Yep. Slipped past me in my testing because I ran the original version once to check the behavior

File: app/util.py (hunk 2 above)
Comment (Trevortds) [reply app/util.py]: I was worried about this, but in my testing, I found that pandas is smart enough to figure out that the nameless column is just an index on its own.

**Comment by Trevortds** [comment]
I think the --keys thing is a good idea, but different enough to be perhaps worth its own issue.

**Comment by Cheukting** [comment]
Thanks @Trevortds I will merge now. I am thinking about having a documentation later and will have a section for all contributors to put their names in. I will give you a shout when it is ready.

**Pull request merged**